src/test/java/org/jenkinsci/plugins/discardbuild/DiscardBuildPublisherTest.java

```diff
@@ -215,6 +237,70 @@ public void testPerformIntervalNumToKeep() throws Exception {
 		verify(buildList.get(20), times(1)).delete(); // to keep
 	}
 
+	public void testPerformHoldMaxBuildsFirstCnd() throws Exception {
+		// testing for circumstance where builds to be discarded
+		// are greater in amount than builds present, causing build discard queue to be cleared
+		DiscardBuildPublisher publisher = getPublisher(new DiscardBuildPublisher(
+				"10", "", "20", "",
+				false, false, false, false, false,
+				"", "", "", false, true));
+
+		// emulates build data and post-build plugin operation
+		publisher.perform((AbstractBuild<?, ?>) buildHMS, launcher, listener);
+
+		for (int i = 1; i < 10; i++) {
```

Comment:
Should test that the running build is not deleted by starting the loop at `int i = 0`.